src/test/java/io/aiven/kafka/connect/s3/S3SinkTaskTest.java

```diff
@@ -538,6 +543,20 @@ final void supportStructValuesForClassicJson() throws IOException {
             testBucketAccessor.readLines("topic1-0-30", compression));
     }
 
+    @Test
+    final void callCredentialProviderFactoryOnStart() {
```

Comment:
I would argue that this test doesn't test anything important

Reply:
It test a that from now on a S3SinkTask delegate Factory to make decision on what Credential Provider to select. I try to use rule - if there is a change, there should be a test.

I could rename test to

`requestCredentialProviderFromFactoryOnStart`